lib/srv/server/azure_watcher.go

```diff
@@ -120,6 +121,7 @@ type azureFetcherConfig struct {
 	Subscription      string
 	ResourceGroup     string
 	AzureClientGetter azureClientGetter
+	DiscoveryConfig   string
```

Comment:
Why is discovery config passed everywhere as a string instead of an object? How are the receivers supposed to use it, parse it every time?

If this is just a name of the config to fetch, then we should rename these fields/variables everywhere, otherwise it's confusing.

Reply:
Renamed everything to `DiscoveryConfigName` and `GetDiscoveryConfigName`